File: RRG/RRG/src/main/java/com/RRG/db/ObjectCreator.java
package com.RRG.db;

import java.sql.ResultSet;
import java.sql.SQLException;

import com.RRG.model.*;

public class ObjectCreator 
{
    public User createUser(ResultSet rs) throws SQLException
    {
        return new User(
            rs.getInt("id"), 
            rs.getString("email"), 
            rs.getString("password")
        );
    }

    public Recipe createRecipe(ResultSet rs) throws SQLException
    {
        return new Recipe(
            rs.getInt("id"),
            rs.getString("name"),
            rs.getString("instructions")
        );
    }
}

File: RRG/RRG/src/main/Python/RRG/db/object_creator.py


class ObjectCreator:
    """Static class for creating ORM objects."""

    @staticmethod
    def create_recipe(name: str, instructions: str = None):
        from model.recipe import Recipe
        return Recipe(name = name, instructions = instructions)
    
    @staticmethod
    def create_ingredient(name:str):
        from model.ingredient import Ingredient
        return Ingredient(name = name)
    
    
